terraform/transform_provider.go

```diff
@@ -117,6 +117,26 @@ func (t *ProviderTransformer) Transform(g *Graph) error {
 
 			// Direct references need the provider configured as well as initialized
 			needConfigured[p.String()] = p
+
+			// attach the provider_meta info
+			if gnapmc, ok := v.(GraphNodeAttachProviderMetaConfigs); ok {
+				log.Printf("[TRACE] ProviderTransformer: attaching provider meta config for %s to %s", p, dag.VertexName(v))
+				if t.Config == nil {
+					log.Printf("[TRACE] ProviderTransformer: no config set on the transformer for %s", dag.VertexName(v))
+					continue
+				}
+				if t.Config.Module == nil {
+					log.Printf("[TRACE] ProviderTransformer: no module in config for %s", dag.VertexName(v))
+					continue
+				}
+				if t.Config.Module.ProviderMetas == nil {
+					log.Printf("[TRACE] ProviderTransformer: no provider metas defined for %s", dag.VertexName(v))
+					continue
+				}
+				if meta, ok := t.Config.Module.ProviderMetas[p.ProviderConfig.Type]; ok {
+					gnapmc.AttachProviderMetaConfigs(meta)
+				}
+			}
```

Comment:
You don't want to attach the metas according to providers, because providers should not be configured in a module. This attaches the provider metas from the module where the provider is configured, which is likely the root module.

Reply:
So wait, `t.Config.Module` is the module where the provider is configured, not the module of the node being transformed?

In my conversations with @apparentlymart we decided on attaching provider meta to resources, and that resources should only have the provider meta for the provider they belong to attached. Is... is that not what this is doing?

Reply:
Providers don't necessarily come from the same module as the resource. They can be inherited or passed into a module explicitly. If the idea is to attach metadata about a module, resources are the only objects that are scoped to a module and communicated to a provider.

What you're doing here will only send module info from where the provider is configured, which should almost never actually be in a module. If you modify the tests here to use modules, you'll see you're getting the wrong metas.